### ISSUE TITLE
borg2 throws error for valid(?) chunker parameters

### ISSUE THREAD

**Issue by agrommek** [borgbackup/borg#7520]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes
## Is this a BUG / ISSUE report or a QUESTION?

A bug, probably

#### Your borg version (borg -V).

borg-2.0.0b5

#### Operating system (distribution) and version.

Arch Linux 

#### Hardware / network configuration, and filesystems used.

Linux, borg-2.0.0b5 precompiled binary, ext4, backup to HDD

#### How much data is handled by borg?

a few GB for testing borg2 beta

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

```
borg-2.0.0b5-linuxnew64 create --chunker-params buzhash,19,24,21,4095 MyArchive /path/to/source
```
(I didn't try this with borg-1.2.x)

## Describe the problem you're observing.

tl;dr: running the above command, I get the following error message:

```
borg-2.0.0b5-linuxnew64 create: error: argument --chunker-params: invalid ChunkerParams value: 'buzhash,19,24,21,4095'
```
Note the value of 24 for CHUNK_MAX_EXP.

Long story: I have a backup source with only large (several hundred MiB each) and seldom-changing files and wanted to adjust the chunker parameters for that by going from the default `buzhash,19,23,21,4095` (512 kiB min, 8 MiB max, 2 MiB avg chunk size) to twice that size (i.e. 1 MiB / 16 MiB / 4 MiB)  with `buzhash,20,24,22,4095` (and maybe even to still larger chunk sizes). However, this throws the error message given above. Some try-and-error revealed that the CHUNK_MAX_EXP parameter is to blame. Selecting anything larger than 23 (i.e. 8 MiB)  for CHUNK_MAX_EXP will trigger the error.

I could not find out why `--chunker-params buzhash,20,24,22,4095` should be invalid. The documentation says nothing about limits and/or inter-dependencies for the chunker values. 

Is this a bug? Maybe a leftover from the "old attic times" where nobody could imagine chunk sizes > 8 MiB?

**Comment by agrommek** [comment]
If CHUNK_MAX_EXP=23 is fixed and the maximum, I can live with that. :-)

But then shouldn't it be documented somewhere (not only in source code) that CHUNK_MAX_EXP is fixed / has a hard limit?  [The docs](https://borgbackup.readthedocs.io/en/master/internals/data-structures.html#chunker-details) don't say anything about hard-codes maximum values. At least the error message could be a bit more detailed to not leave people in the dark what they did wrong, no?

**Comment by ThomasWaldmann** [comment]
I could imagine that maybe 24 (16MiB) could also still work, but guess nobody tried and reviewed the code for that yet.

About the error msg: master branch code has this:
```
    if algo == CH_BUZHASH and count == 5 or count == 4:  # [buzhash, ]chunk_min, chunk_max, chunk_mask, window_size
        chunk_min, chunk_max, chunk_mask, window_size = (int(p) for p in params[count - 4 :])
        if not (chunk_min <= chunk_mask <= chunk_max):
            raise ValueError("required: chunk_min <= chunk_mask <= chunk_max")
        if chunk_min < 6:
            # see comment in 'fixed' algo check
            raise ValueError("min. chunk size exponent must not be less than 6 (2^6 = 64B min. chunk size)")
        if chunk_max > 23:
            raise ValueError("max. chunk size exponent must not be more than 23 (2^23 = 8MiB max. chunk size)")
        return CH_BUZHASH, chunk_min, chunk_max, chunk_mask, window_size
```

**Comment by ThomasWaldmann** [comment]
Strange, that code somehow does not work as expected, I just reproduced it.

**Comment by jdchristensen** [comment]
This line should probably have parentheses to make the intended logic clear:
```
if algo == CH_BUZHASH and count == 5 or count == 4:
```

**Comment by agrommek** [comment]
> This line should probably have parentheses to make the intended logic clear:
> 
> ```
> if algo == CH_BUZHASH and count == 5 or count == 4:
> ```

I agree... I had to look up operator precedence in the [Python docs](https://docs.python.org/3/reference/expressions.html#operator-precedence)... for the record: `and` binds stronger than `or`, so the logic is correct.

**Comment by agrommek** [comment]
Just an additional data point.... the [Python docs for argparse](https://docs.python.org/3/library/argparse.html#type) say the following:

> The argument to type can be any callable that accepts a single string. If the function raises ArgumentTypeError, [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), or [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError), the exception is caught and a nicely formatted error message is displayed. No other exception types are handled.

So ValueError should work and be ok?

**Comment by ThomasWaldmann** [comment]
No, only `argparse.ArgumentTypeError` gives the nice error msg. I'll fix it.
